github/prci.py

```diff
@@ -19,11 +22,76 @@
 )
 from internals.gql import util, queries
 
-
 logger = logging.getLogger(__name__)
 
 
 ERROR_BACKOFF_TIME = 600
+OS_PID_FILE_PATH = "/run/"
+OS_PID_FILE = os.path.join(OS_PID_FILE_PATH, "prci.pid")
+
+
+def pid_file_creation():
+    """
+        Creates PID file with Proccess ID as content.
+    """
+    with open(OS_PID_FILE, "w") as pid_file:
+        pid_file.write(str(os.getpid()))
+
+
+def pid_file_handling():
+    """
+        Create the PID file for each run.
+    """
+    if os.path.exists(OS_PID_FILE):
+        vagrant_process_prune()
+        vagrant_machines_prune()
```

Comment:
`vagrant_machines_prune()` calls `vagrant_process_prune()`, so the latter is called twice.

Reply:
`vagrant_process_prune` is called in context of `machine_ prune` to ensure that there is no process locking vm's for being destroyed.

If any process rise in the middle term or if any call is made straight to `machine_prune` it will fail to prune the vms.

Reply:
is it called twice because is needed? If so, make it more explicit, this type of change in the flow is always a bit confusing.